pkg/tests/non-dependant/smcp_install_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package non_dependant
+
+import (
+	_ "embed"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/tests/ossm"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+var (
+	smmr             = ossm.GetSMMRTemplate()
+	versionTemplates = ossm.GetSMCPTemplates()
+)
+
+func TestSMCPInstall(t *testing.T) {
+	NewTest(t).Id("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+		vars := map[string]string{
+			"Name":      smcpName,
+			"Namespace": meshNamespace,
+			"Rosa":      strconv.FormatBool(env.IsRosa()),
+		}
+		smcpVersion := env.GetDefaultSMCPVersion()
+		_, ok := versionTemplates[smcpVersion]
+		if !ok {
+			t.Fatal("Unsupported SMCP version: " + smcpVersion)
+		}
+		for version, smcpTemplate := range versionTemplates {
```

Comment:
Instead of iterating over the map directly, I think it maybe be better to use the `version` variable that you define a few lines below. This way, the order of entries in the map won't affect this test.

Reply:
I'd also add an empty line above and below this loop, so that it stands out as a separate piece of code (this piece tests the installation, whereas the next piece tests the upgrade).